### PR TITLE
Sea level rise pipeline

### PR DESCRIPTION
**Pull request by mastaia**
This PR contains the integration of the sea level rise hazard into the pipeline

**get hazard from s3 bucket:** 
- currently, we use the following files:
- historical climate, no SLR: 
            tc_surge/no_cc/no_slr/
- historical climate, future SLR: 
            tc_surge/no_cc/ssp126_2060slr/
            tc_surge/no_cc/ssp585_2060slr/

**Config**
- contains a new config file to run the sea level hazard: nccs/run_configurations/config_sea_level.py

**Calculate change between baseline and future**
- there is a separate script that needs to be executed after the sea level pipeline run and the csv flies are created
- run name needs to be indicated
- it is called: nccs/sea_level_rise_difference.py
- It creates a new folder in the indirect path of the run where for each country, sector, and io approach two files get produced: Historical vs ssp126 2060 and historical vs ssp585 2060
- Future minus historical --> Positive values indicate more impact in the future, negative values, less impact in the future

### CODE REVIEW

File: nccs/pipeline/direct/direct.py [hunk 1]
```diff
@@ -426,6 +438,31 @@ def get_hazard(haz_type, country_iso3alpha, scenario, ref_year):
         )
         return haz
 
+    elif haz_type == 'sea_level_rise':
+        """
+        Sea level rise has the following file configurations:
+        
+        "future climate" includes changes also to the TC surge
```
Comment (ChrisFairless) [inline nccs/pipeline/direct/direct.py:445]:
```suggestion
        "future climate" includes changes also to the TC surge
        Note that (currently) this method always returns historical TC event sets with 
        sea level rise applied according to the requested scenario.
```

File: nccs/sea_level_rise_difference.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+
+# Ensure the output folder exists
+os.makedirs(output_folder, exist_ok=True)
+
+def parse_filename(filename):
+    """
+    Parse the filename into its components: sector, scenario, year, methodology, country
+    Assumes filenames follow this structure:
+    indirect_impacts_sea_level_rise_<hazard>_<sector>_<scenario>_<year>_<methodology>_<country>.csv
```
Comment (ChrisFairless) [inline nccs/sea_level_rise_difference.py:34]:
```suggestion
    indirect_impacts_sea_level_rise_<sector>_<scenario>_<year>_<methodology>_<country>.csv
```

**Comment by ChrisFairless** [comment]
Looks great! Thanks for building this!

It runs well on my local machine and I get outputs. The changes are ... very small for the country/sector combinations I looked at. Does that match what you're seeing?

Also, it looks like there are a bunch of non-sea-level changes included in the PR. I didn't look at them!

**Comment by mastaia** [comment]
Hey Chris, 

Thanks a lot for the review!

Yes, the changes in impact were also fairly small for me

Great, then I will merge it and no worries about the non-related sea level commits. When going through the exposure I just did some beauty corrections since we are at the end of the project. ;)

**Pull request merged**